packages/xo-server/src/xapi/index.js

```diff
@@ -343,6 +343,25 @@ export default class Xapi extends XapiBase {
     await this.call('host.enable', this.getObject(hostId).$ref)
   }
 
+  async installCertificateOnHost(
+    hostId,
+    { certificate, certificateChain = '', privateKey }
+  ) {
+    try {
+      await this.call(
+        'host.install_server_certificate',
+        this.getObject(hostId).$ref,
+        certificate,
+        privateKey,
+        certificateChain
+      )
+    } catch (error) {
+      if (error.code !== 'ECONNRESET') {
+        throw error
+      }
```

Comment:
Yes, CH/XCP-ng resets the connection to this host after a certificate install (I hesitated to add a comment on this line because i thought the mean of the error is clear).